### PR TITLE
Empty Theme: Script to generate a personalized starter theme

### PR DESCRIPTION
**Pull request by MaggieCabrera**
Using https://github.com/Automattic/underscores.me/blob/master/plugins/underscoresme-generator/underscoresme-generator.php as a reference I created a script that will generate a zip file with a theme based on `emptytheme` that has the name the user gives to the script.

Run `php new-empty-theme.php` from the root of the repo.

We can expand on this to add further info for style.css such as Author, Description, etc. I only did the name as a starting point.

### CODE REVIEW

File: new-empty-theme.php [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+	function create_zip() {
+
+		if($this->theme['name'] === ''){
+			echo "ABORTING!\n";
+			exit;
+		}
+
+		$zip = new ZipArchive;
+		$zip_filename = sprintf( '%s.zip', md5( print_r( $this->theme['name'], true ) ) );
```
Comment (jffng) [inline new-empty-theme.php:29]: This results in a zip with a name as an md5 hash e.g. `098f6bcd4621d373cade4e832627b4f6.zip`. Is this intentional?

**Review by jffng — COMMENTED** [review]
Looking good and working well for me!

We should add some documentation to the readme about the php requirement and how to use this.

File: new-empty-theme.php (hunk 1 above)
Comment (mendezcode) [reply new-empty-theme.php:29]: I think the slug would be better suited for the theme name, e.g. `my-theme.zip` instead of the MD5.

**Comment by scruffian** [comment]
Is it more convenient for this to generate a zip or a directory?

File: new-empty-theme.php (hunk 1 above)
Comment (MaggieCabrera) [reply new-empty-theme.php:29]: yeah, it's the approach by the original underscores' script but we can change this definitely

**Comment by MaggieCabrera** [comment]
> Is it more convenient for this to generate a zip or a directory?

I am not 100% sure of how we want to go with this script and that's why I kept the approach of the underscores original code. And also why I didn't add any extra info for the style.css.

I think this is a good starting point and the next steps would be:

- [ ] Changing from generating a zip file to just a folder
- [ ] Ask the user for the author name, theme URL, and description. (Do we have a default in case the user doesn't fill one of these?) 
- [ ] Add documentation for this script in the README

Does that sound good? Anything else we could add?

**Comment by carolinan** [comment]
So happy to see work on this starting 🎉

**Comment by MaggieCabrera** [comment]
This is ready for another review :D

File: new-empty-theme.php [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		echo "Author: ";
+		$input = fopen ("php://stdin","r");
+		$this->theme['author'] = trim(fgets($input));
+
+		echo "Theme URI: ";
+		$input = fopen ("php://stdin","r");
+		$this->theme['uri'] = trim(fgets($input));
+	}
+
+	function create_zip() {
```
Comment (scruffian) [inline new-empty-theme.php:51]: We should probably change the name of this function!

**Review by scruffian — APPROVED** [review]
LGTM

**Comment by scruffian** [comment]
In style.css I still see this:

`Empty Theme is distributed under the terms of the GNU GPL.`

**Comment by scruffian** [comment]
The block template parts also have the theme slug in them.

**Comment by MaggieCabrera** [comment]
> The block template parts also have the theme slug in them.

I think the theme slug should be removed from the template parts like Jorge says here: https://github.com/WordPress/theme-experiments/pull/81#discussion_r531186733

**Review by scruffian — APPROVED** [review]
LGTM

**Comment by kjellr** [comment]
🎉 Thanks for this, @MaggieCabrera!

**Pull request merged**